lib/perl/Genome/Sys/Lock.pm

```diff
@@ -97,6 +97,8 @@ sub lock_resource {
 
     $class->_cleanup_handler_check();
 
+    $RESOURCE_LOCK_SCOPE{$args{resource_lock}} = $args{scope};
```

Comment:
This is basically the same concern Adam expressed.  I'm still trying to think about it.